### PR TITLE
feat(providers): yaml content

### PR DESCRIPTION
**Pull request by eddybrando**
## Changes

This PR makes it possible to fetch the providers data from YAML files instead of JSON using the [Nuxt Content module](https://content.nuxtjs.org/).

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Eddy! Just some small modifications on naming and typing

File: types/providersContent.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import type { ParsedContent } from "@nuxt/content/dist/runtime/types";
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply types/providersContent.ts]: Changed in https://github.com/Qiskit/qiskit.org/pull/3130/commits/ea691ab3fb3e10e7aa0a6ffe2c8acd7f16e855fa

File: components/Landing/LandingQuickStart/LandingQuickStartComponent.vue [hunk 2]
```diff
@@ -35,9 +35,13 @@
...
 <script setup lang="ts">
-import rawQuickStartProvidersData from "~/content/providers/quick-start.json";
+import type { ProviderObject } from "~/types/providersContent";
 
-const providersData = ref(rawQuickStartProvidersData);
+const { data } = await useAsyncData("providers-quick-start", () =>
+  queryContent("/providers/quick-start/data").findOne()
+);
+
+const providersData = data.value!.body as ProviderObject[];
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply components/Landing/LandingQuickStart/LandingQuickStartComponent.vue]: Changed in https://github.com/Qiskit/qiskit.org/pull/3130/commits/b7d00de25b2859c4b54b3b08ad095a2b4f029da7

File: pages/providers.vue [hunk 3]
```diff
@@ -70,8 +70,21 @@ useHead({
   title: "Qiskit Providers",
 });
 
-const tocEntries = TABLE_OF_CONTENTS;
-const contentSections = CONTENT_SECTIONS;
+const { data: providersData } = await useAsyncData("providers", () =>
+  queryContent("/providers/list").find()
+);
+
+const contentSections = providersData.value as ProvidersSectionParsedContent[];
```
[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/providers.vue]: Changed in https://github.com/Qiskit/qiskit.org/pull/3130/commits/b7d00de25b2859c4b54b3b08ad095a2b4f029da7

[the remark this answers is not in the document]
Comment (eddybrando) [reply pages/providers.vue]: Sorry, changed again in https://github.com/Qiskit/qiskit.org/pull/3130/commits/b9fa6038323c239dc43a000faa1b689fe29248d9

File: components/Landing/LandingQuickStart/LandingQuickStartComponent.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (eddybrando) [reply components/Landing/LandingQuickStart/LandingQuickStartComponent.vue]: Sorry, changed again in https://github.com/Qiskit/qiskit.org/pull/3130/commits/b9fa6038323c239dc43a000faa1b689fe29248d9

**Review by y4izus — APPROVED** [review]
Thank you Eddy!

**Pull request merged**